File: libs/collection/src/main/java/org/toolkit4j/collection/tree/SetTreeNode.java
package org.toolkit4j.collection.tree;

import java.util.Set;

/**
 * 子节点为 Set 的树节点，兄弟无重复。
 * TreeNode with unique children (Set).
 *
 * @param <T> 节点数据类型
 */
public interface SetTreeNode<T> extends TreeNode<T> {

  @Override
  Set<TreeNode<T>> children();
}

File: kit/collection/src/main/kotlin/org/toolkit4j/collection/tree/SetTreeNode.kt
package org.toolkit4j.collection.tree

class SetTreeNode<T>
@JvmOverloads
constructor(
    override var data: T?,
    override var parent: TreeNode<T>? = null,
    override val children: MutableSet<TreeNode<T>?> = mutableSetOf(),
) :
    AbstractTreeNode<T>(data, parent) {
}